lldb/include/lldb/Interpreter/CommandInterpreter.h

```diff
@@ -766,6 +768,12 @@ class CommandInterpreter : public Broadcaster,
   CommandUsageMap m_command_usages;
 
   StreamString m_transcript_stream;
+
+  /// Contains a list of handled commands, output and error. Each element in
+  /// the list is a dictionary with three keys: "command" (string), "output"
+  /// (list of strings) and optionally "error" (list of strings). Each string
+  /// in "output" and "error" is a line (without EOL characters).
+  StructuredData::ArraySP m_transcript;
```

Comment:
This shouldn't be a shared pointer, we don't want to hand out references to this structured data across the API boundary because it can be accessed by the caller and also be updated by new commands coming in. This should be a `StructuredData::Array` object and we should copy it when we hand it out across the API boundary.

Reply:
Or we can leave this as a shared pointer and when  people call GetTranscript() on this object we give away our copy and start a new one.

Reply:
Not sure if this would make sense to the user though...

Reply:
Improve the loop as discussed (I think).